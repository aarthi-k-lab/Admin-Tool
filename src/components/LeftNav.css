.leftmenubar {
  list-style-type: none;
  margin: -8px;
  padding: 0;
  width: 60px;
  height: 100vh;
  background-color: Black;
}

.leftmenu {
  margin-left: 10px;
  margin-top: 18px;
}
