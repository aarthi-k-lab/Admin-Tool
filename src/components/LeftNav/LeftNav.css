.left-nav-bar {
  align-items: center;
  background-color: black;
  display: inline-flex;
  flex-direction: column;
  flex-grow: 1;
  list-style-type: none;
  width: 4rem;
}

.left-nav-bar > * {
  color: #fff;
  cursor: pointer;
  font-size: 2rem;
  padding: 1rem;
  width: 4rem;
}

.left-nav-bar > * > * {
  width: 2rem;
}

.left-nav-bar > *:hover {
  background-color: #3d4658;
}

.active-bar {
  background-color: #3d4658;
}

.stretch-column {
  display: flex;
  flex-direction: column;
}
