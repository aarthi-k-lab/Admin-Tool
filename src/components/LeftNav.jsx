import React from 'react';

import './LeftNav.css';
import Dashboard from '../images/Dashboard.png';

function LeftNav() {
  return (
    <div id="cmod_leftnav" styleName="leftmenubar">
      <img alt="menu" src={Dashboard} styleName="leftmenu" />
    </div>
  );
}

export default LeftNav;
