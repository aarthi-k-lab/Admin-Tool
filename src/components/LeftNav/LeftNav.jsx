import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './LeftNav.css';

const links = [
  {
    path: '/reports',
    name: 'dashboard',
    img: '/static/img/icon-dashboard.png',
    groups: ['feuw-mgr', 'beuw-mgr'],
  },
  {
    path: '/loan-evaluation',
    name: 'loan-evaluation',
    img: '/static/img/frontend.svg',
    groups: ['feuw', 'feuw-mgr'],
  },
  {
    path: '/stager',
    name: 'stager',
    img: '/static/img/stager.svg',
    groups: ['feuw-mgr', 'beuw-mgr', 'stager', 'stager-mgr'],
  },
  {
    path: '/move-forward',
    name: 'move-forward',
    img: '/static/img/move_forward.svg',
    groups: ['feuw-mgr', 'beuw-mgr'],
  },
];

function LeftNav({ user }) {
  const groupList = user && user.groupList;
  return (
    <div styleName="stretch-column">
      <nav id="cmod_leftnav" styleName="left-nav-bar">
        {
        links.map(link => (
          groupList && groupList.some(r => link.groups.includes(r))
            ? (
              <Link to={link.path}>
                <img alt={link.name} src={link.img} />
              </Link>) : null
        ))
      }
      </nav>
    </div>
  );
}

LeftNav.propTypes = {
  user: PropTypes.shape({
    userDetails: PropTypes.shape({
      email: PropTypes.string,
      jobTitle: PropTypes.string,
      name: PropTypes.string,
    }),
    userGroups: PropTypes.array,
  }).isRequired,
};

export default LeftNav;
