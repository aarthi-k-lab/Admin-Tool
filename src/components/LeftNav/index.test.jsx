import React from 'react';
import renderer from 'react-test-renderer';
import LeftNav from './LeftNav';

describe('<LeftNav Components />', () => {
  it('LeftNav renders correctly', () => {
    const user = {
      userDetails: {
        email: 'bernt@mrcooper.com',
        jobTitle: 'CEO',
        name: 'brent',
      },
    };
    const tree = renderer
      .create(<LeftNav user={user} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
