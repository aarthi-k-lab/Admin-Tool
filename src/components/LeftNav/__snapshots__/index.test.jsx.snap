// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftNav Components /> LeftNav renders correctly 1`] = `
<div
  className="components-LeftNav-LeftNav__stretch-column--2Cozo"
>
  <nav
    className="components-LeftNav-LeftNav__left-nav-bar--3fFXS"
    id="cmod_leftnav"
  />
</div>
`;
